### DISCUSSION TITLE
FreeRTOS Readme and examples needs to update?

### DISCUSSION THREAD

**Discussion by board707** [earlephilhower/arduino-pico#2160] [Q&A] [answered]
Hi

It seems that FreeRTOS library (`libraries/FreeRTOS`) Readme description is outdated. For example, it startes with the requirement, that the file `RP2040_FreeRTOS.h` must be included in the sketch first, while there is no such file in the library at all.

As about the examples, the one `MultiCore-FreeRTOS` seems to work, but `staticMultiCore-FreeRTOS` doesn't. If you take the latter code, and load it into the board - the LED lights up and does not blink, and the serial port freezes and the board is no longer detected by Windows.

**Comment by earlephilhower** [comment, marked as answer]
The README.md was untouched from the original port-of-a-port so please ignore anything it says. :)

For the example, it does work but it's got race conditions such that as soon as a task gives up a mutex it re-grabs it before the other task can take it, so the led doesn't blink.  Adding a `delay(1)` to the end of each `while` takes care of this, so I'll throw a PR up for it.  Also change the default output to the USB port, not UART0.

**Comment by board707** [comment]
Thank you very much, after adding the delay the example works fine.

Addition.
Also I think that the test should be named `staticSingleCore-FreeRTOS`, because it seems do not use a second core...
